src/api.js

```diff
@@ -0,0 +1,114 @@
+var _ = require('underscore');
+var createDashboard = require('./create-dashboard');
+var cdb = require('cartodb.js');
+
+function Dashboard (dashboard) {
+  this.dashboard = dashboard;
+}
+
+Dashboard.prototype = {
+
+  /**
+   * @return {Map} the map used in the dashboard
+   */
+  getMap: function () {
+    return this.dashboard.vis;
+  },
+
+  /**
+   * @return {Array} of widgets in the dashboard
+   */
+  getWidgets: function () {
+    return this.dashboard.widgets.getList();
+  },
+
+  /**
+   * @param {Integer} id - widget id
+   * @return a widget object
+   */
+  getWidget: function (id) {
+    return this.dashboard.widgets.get(id);
+  },
+
+  /**
+   * Create a category widget.
+   * @param {Object} widgetAttrs - attributes for the new widget
```

Comment:
I'm missing some important attributes here. eg: `column` for the category widget or bins for the histogram widget. It looks like common attributes for all widgets are documented here, but some other important ones are not. That's what I wanted to say with "are all the same".